packages/db/src/provider.ts

```diff
@@ -0,0 +1,59 @@
+import omit from 'lodash/omit';
+import isNil from 'lodash/isNil';
+import isValidHostname from 'is-valid-hostname';
+
+import { ProviderRecord, PostgresProviderRecord } from '@cumulus/types';
+import { PostgresValidationError } from '@cumulus/errors';
+
+import KMS from '@cumulus/aws-client/KMS';
+
+export const encryptValueWithKMS = (
+  value: string
+): Promise<string> => {
+  if (process.env?.provider_kms_key_id === undefined) {
+    throw new Error('env variable provider_kms_key_id must be set');
+  }
+  return KMS.encrypt(process.env?.provider_kms_key_id, value);
+};
+
+export const postgresProviderFromCumulusProvider = async (
```

Comment:
Super nit: sometimes I'm naming these kind of methods using `RDS` (e.g. `shouldWriteExecutionToRDS`) vs "Postgres". It'd probably be good to standardize one way or the other

Reply:
I kinda like `dynamoProviderToPostgresProvider` or even `translateDynamoProviderToPostgres`. I feel like `CumulusProvider` is ambiguous

Reply:
Agree re: the nit - I've been trying to settle on Postgres as we're not actually constrained to RDS.

Reply:
I like `translateDynamoProviderToPostgres` since it's a verb - just my two cents 🔅

Reply:
I'm not really a fan of using Dynamo here, given Dynamo is going away, but the concept of a Cumulus API Provider object may not be. 

I'll concede that point for the moment, so long as we're comfortable with eventually refactoring anything using that convention if we retain it once Dynamo is gone.

Reply:
> I'm not really a fan of using Dynamo here, given Dynamo is going away, but the concept of a Cumulus API Provider object may not be.

Hmm, but at the point where the Cumulus API stops returning "Dynamo"-style records, it'll be because we've switched the API to read from RDS, so it still seems valid?

I guess your point about the "Cumulus API provider object" is that we still need translation logic until we update the expected request body for these endpoints?

Regardless, I agree with @botanical that something with "translate" in the name is best.

Reply:
> Hmm, but at the point where the Cumulus API stops returning "Dynamo"-style records

The point is that switching to reads doesn't necessarily mean that we're updating the API to use Postgres-style records, or at least it's not clear to me that that's the case.

Reply:
Right, which is what I was trying to get at by saying this

> I guess your point about the "Cumulus API provider object" is that we still need translation logic until we update the expected request body for these endpoints?

I guess I'm fine with `translateCumulusProviderToPostgres`

Reply:
chatted via hangouts, and think we've settled on translateApi<thing>toPostgres<thing>.   Will elaborate in slack.

Reply:
Uodated in c591c12ea